src/algo/mod.rs

```diff
@@ -918,3 +919,148 @@ impl FloatMeasure for f64 {
         1. / 0.
     }
 }
+
+pub trait BoundedMeasure: Measure + std::ops::Sub<Self, Output=Self> {
```

Comment:
You could replace this with a Macro:

	pub trait BoundedMeasure: Measure + std::ops::Sub<Self, Output=Self> {
	    fn min() -> Self;
	    fn max() -> Self;
	    fn overflowing_add(self, rhs: Self) -> (Self, bool);
	}


	macro_rules! impl_bounded_measure_integer(
	    ( $( $t:ident ),* ) => {
		$(
		    impl BoundedMeasure for $t {
		        fn min() -> Self {
		            std::$t::MIN
		        }
		    
		        fn max() -> Self {
		            std::$t::MAX
		        }
		        
		        fn overflowing_add(self, rhs: Self) -> (Self, bool) {
		            self.overflowing_add(rhs)
		        }
		    }   
		)*
	    };
	);

	impl_bounded_measure_integer!(u8, u16, u32, u64, u128, usize, i8, i16, i32, i64, i128, isize);

	macro_rules! impl_bounded_measure_float(
	    ( $( $t:ident ),* ) => {
		$(
		    impl BoundedMeasure for $t {
		        fn min() -> Self {
		            std::$t::MIN
		        }
		    
		        fn max() -> Self {
		            std::$t::MAX
		        }
		        
		        fn overflowing_add(self, rhs: Self) -> (Self, bool) {
		            // for an overflow: a + b > max: both values need to be positive and a > max - b must be satisfied
		            let overflow = self > Self::default() && rhs > Self::default() && self > std::$t::MAX - rhs;
	    
		            // for an underflow: a + b < min: overflow can not happen and both values must be negative and a < min - b must be satisfied
		            let underflow = !overflow && self < Self::default() && rhs < Self::default() && self < std::$t::MIN - rhs;
	    
		            (self + rhs, overflow || underflow)
		        }
		    }   
		)*
	    };
	);

	impl_bounded_measure_float!(f32, f64);